src/schema/posts.ts

```diff
@@ -1101,6 +1223,129 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
   BaseContext
 >({
   Query: {
+    sourcePostModeration: async (
+      _,
+      { id, sourceId }: { id: string; sourceId: string },
+      ctx: Context,
+      info,
+    ): Promise<GQLSourcePostModeration> => {
+      const isModerator = await isPrivilegedMember(ctx, sourceId);
+
+      return graphorm.queryOneOrFail<GQLSourcePostModeration>(
+        ctx,
+        info,
+        (builder) => {
+          const queryBuilder = builder.queryBuilder.where(
+            `"${builder.alias}"."id" = :id AND ${builder.alias}."sourceId" = :sourceId`,
+            { id, sourceId },
+          );
+
+          if (!isModerator) {
+            queryBuilder.andWhere(
+              `"${builder.alias}"."createdById" = :userId`,
+              {
+                userId: ctx.userId,
+              },
+            );
+          }
+          return {
+            ...builder,
+            queryBuilder,
+          };
+        },
+      );
+    },
+    sourcePostModerations: async (
+      _,
+      args: SourcePostModerationArgs,
+      ctx: Context,
+      info,
+    ): Promise<ConnectionRelay<GQLSourcePostModeration>> => {
+      const isModerator = await isPrivilegedMember(ctx, args.sourceId);
+      const page = sourcePostModerationPageGenerator.connArgsToPage(args);
+
+      if (isModerator) {
+        return graphorm.queryPaginated(
+          ctx,
+          info,
+          (nodeSize) =>
+            sourcePostModerationPageGenerator.hasPreviousPage(page, nodeSize),
+          (nodeSize) =>
+            sourcePostModerationPageGenerator.hasNextPage(page, nodeSize),
+          (node, index) =>
+            sourcePostModerationPageGenerator.nodeToCursor(
+              page,
+              args,
+              node,
+              index,
+            ),
+          (builder) => {
+            builder.queryBuilder
+              .where(`"${builder.alias}"."sourceId" = :sourceId`, {
+                sourceId: args.sourceId,
+              })
+              .orderBy(`${builder.alias}.updatedAt`, 'DESC')
+              .limit(page.limit)
+              .offset(page.offset);
+
+            if (args.status) {
+              builder.queryBuilder.andWhere(
+                `"${builder.alias}"."status" IN (:...status)`,
+                {
+                  status: Array.from(new Set(args.status)),
+                },
+              );
+            }
```

Comment:
Maybe we want to set `pending` as default status, wdyt?

Reply:
I think if no status is set we don't query on status

Reply:
if no statuse were sent then the result would always be empty, because `Array.from(new Set(undefined))` is `[]`